src/sage/manifolds/chart.py

```diff
@@ -553,7 +602,7 @@ def _latex_(self):
         description += latex(self._xx[n-1]).strip() + r')\right)'
         return description
 
-    def _first_ngens(self, n):
+    def _first_ngens(self, n: int):
```

Comment:
`int` is not correct; here and in many other places. A Sage `Integer` is definitely allowed. 
I'd suggest to use the numeric ABCs; https://docs.python.org/3/library/numbers.html

Reply:
these abc's are not supported (in the way you want) by pyright and mypy, see https://github.com/microsoft/pyright/issues/1575 and references therein. We also cannot use `Integer` since its a cython class and there are no stubs for that (yet). So yeah, its a partial information that is still better than nothing (since ints are allowed).

Reply:
I'm pretty sure that the mantra "don't lie to your static checker" should apply.

Acceptable "partial information" would be using a supertype, not a subtype, of what is allowed. If our `Integer` cannot be represented, then this will just have to be an `Any` or `object`.

Reply:
@tobiasdiez Do you have an opinion on what tools to use for generating the typing stubs (pyi) for our Cython classes? A quick search led me to https://github.com/RaubCamaioni/CythonPEG, but I'm guessing there must be more available

Reply:
> Acceptable "partial information" would be using a supertype, not a subtype, of what is allowed. If our `Integer` cannot be represented, then this will just have to be an `Any` or `object`.

With `any` you don't get an error if you pass in a string. With `int` you can pass in any sage integer (which is treated as any). Since there are so many open challenges with even getting basic type checks to work, the mantra would be more "over time lie less and less to your static checker".

> @tobiasdiez Do you have an opinion on what tools to use for generating the typing stubs (pyi) for our Cython classes? A quick search led me to https://github.com/RaubCamaioni/CythonPEG, but I'm guessing there must be more available

There is some movement to implement this directly in Cython. I would wait for this. There is an open issue in the Cython repo (that also compares different workarounds/packages in case you are interested)...

Reply:
> > Acceptable "partial information" would be using a supertype, not a subtype, of what is allowed. If our `Integer` cannot be represented, then this will just have to be an `Any` or `object`.

+1 

> With `any` you don't get an error if you pass in a string. With `int` you can pass in any sage integer (which is treated as any). Since there are so many open challenges with even getting basic type checks to work, the mantra would be more "over time lie less and less to your static checker".

It's not only that you are lying to your static checker: you are also lying to the user/developer examining the source code. IMHO, no typing information is better than a false one.

Reply:
Note that the objective of these type checks is to discover bugs. In this case, that you get notified about passing something wrong into the method. The idea that the types should strictly represent the runtime behavior is not very helpful. The numpy docs for example state
> NumPy is very flexible. Trying to describe the full range of possibilities statically would result in types that are not very helpful. For that reason, the typed NumPy API is often stricter than the runtime NumPy API.

But yes, I agree that finding a solution for the integers would be nice. Let me think about it (and propose ideas in case you have some).